source/parse.h

```diff
@@ -3318,6 +3322,24 @@ class parser
                 }
             }
 
+            if (deduced_type) {
+                auto& type = std::get<declaration_node::object>(n->type);
+                if (peek(1)->type() == lexeme::Ampersand) {
+                    type->address_of = &curr();
+                }
```

Comment:
So we peeking at the first character after the `n->equal_sign`, at the second token of the initializer? Looking for initializers of the form `var&` only if they're unqualfied -- correct?

Reply:
Yes, on the unqualified.